src/documentation/literal.rs

```diff
@@ -0,0 +1,443 @@
+            (cutoff.apply_to(&data[..start]).to_string(), String::new())
+        } else {
+            (String::new(), "!!!".to_owned())
+        };
+        // highlight the given range
+        let highlight = if end >= data.len() {
+            oob.apply_to(&data[start..data.len()]).to_string()
+        } else {
+            highlight.apply_to(&data[start..end]).to_string()
+        };
+        // color trailing context if any as well as the closing quote character
+        let post_idx = literal.pre() + literal.len();
+        let (ctx2, post) = if post_idx > end {
+            (
+                context.apply_to(&data[end..post_idx]).to_string(),
+                cutoff.apply_to(&data[post_idx..]).to_string(),
+            )
+        } else if end < data.len() {
+            (String::new(), cutoff.apply_to(&data[end..]).to_string())
+        } else {
+            (String::new(), oob.apply_to("!!!").to_string())
+        };
+
+        write!(formatter, "{}{}{}{}{}", pre, ctx1, highlight, ctx2, post)
+    }
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+    use crate::LineColumn;
+
+    pub(crate) fn annotated_literals(source: &str) -> Vec<TrimmedLiteral> {
+        let stream =
+            syn::parse_str::<proc_macro2::TokenStream>(source).expect("Must be valid rust");
+        stream
+            .into_iter()
+            .filter_map(|x| {
+                if let proc_macro2::TokenTree::Group(group) = x {
+                    Some(group.stream().into_iter())
+                } else {
+                    None
+                }
+            })
+            .flatten()
+            .filter_map(|x| {
+                if let proc_macro2::TokenTree::Literal(literal) = x {
+                    Some(literal)
+                } else {
+                    None
+                }
+            })
+            .map(|literal| {
+                TrimmedLiteral::try_from(literal)
+                    .expect("Literals must be convertable to trimmed literals")
+            })
+            .collect()
+    }
+
+    const PREFIX_RAW_LEN: usize = 3;
+    const SUFFIX_RAW_LEN: usize = 2;
+    const GAENSEFUESSCHEN: usize = 1;
+
+    struct Triplet {
+        /// source content
+        source: &'static str,
+        /// expected doc comment content without modifications
+        extracted: &'static str,
+        /// expected doc comment content after applying trimming rules
+        trimmed: &'static str,
+        /// expected span as extracted by proc_macro2
+        extracted_span: Span,
+        /// trimmed span, so it is aligned with the proper doc comment
+        trimmed_span: Span,
+    }
+
+    const TEST_DATA: &[Triplet] = &[
+        Triplet {
+            source: r#"
+/// One Doc
+struct One;
+"#,
+            extracted: r#"" One Doc""#,
+            trimmed: " One Doc",
+            extracted_span: Span {
+                start: LineColumn {
+                    line: 2usize,
+                    column: 0, // @todo why????
```

Comment:
that's still not clear right?

Reply:
I think it's a quirk of `proc_macro2`, it's just what you get with `/// doc comment`, but the `expacted_spans` are not checked against any more and the quirk is applied in `impl From<proc_macro2::Literal> for TrimmedLiteral`, so it should be of no concern in this test anymore.